docs/explanation/cryptography.rst

```diff
@@ -0,0 +1,303 @@
+more information, see the  `Craft Application cryptography`_ documentation.
+
+Downloading build dependencies
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:ref:`Plugins <plugins>` use build tools to download and verify build dependencies. Some
+plugins can provision their own build tools, while others require the build tools to be
+available on the system
+
+For more information on the use of cryptography for plugins provided by Craft Parts, see
+the `Craft Parts cryptography`_ documentation.
+
+For plugins provided by Snapcraft, the table below summarizes how plugins provision
+build tools and which build tools are used to download and verify dependencies.
+
+.. list-table::
+  :header-rows: 1
+
+  * - Plugin
+    - Build tools used
+    - Method of provisioning the build tools
+
+  * - :ref:`Colcon <reference_colcon_plugin>`
+    - ``colcon`` and ``rosdep``
+    - not provisioned
+
+  * - Conda
+    - ``conda``
+    - Requests library and `curl`_
+
+  * - Flutter
+    - ``flutter``
+    - Git
+
+  * - :ref:`Matter SDK <reference_matter_sdk_plugin>`
+    - ``matter``
+    - Git
+
+core22
+------
+
+Snaps using core22 as their base snap depend on various craft libraries to build.
+Snapcraft's cryptographic functionality with these libraries is described below.
+
+Public key signing
+~~~~~~~~~~~~~~~~~~
+
+Snapcraft supports the adding and verification of arbitrary package repositories. For
+more information, see the `Craft Archives cryptography`_ documentation.
+
+The parts system
+~~~~~~~~~~~~~~~~
+
+Snapcraft makes use of *parts* in project files for declarative builds. Parts specified
+by the user may download arbitrary files, install packages, and more. For more
+information, see the `Craft Parts cryptography`_ documentation.
+
+Creating virtual build environments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Snapcraft instantiates and executes builds on self-allocated virtual instances. For more
+information, see `Craft Providers cryptography`_ documentation.
+
+Downloading build dependencies
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Snaps using core22 have the same plugin support as core24. See the core24 section above
+for more information on how Snapcraft downloads and verifies build dependencies.
+
+core20
+------
+
+Snaps using core20 as their base snap depend on various craft libraries to build.
+Snapcraft's cryptographic functionality with these libraries is described below.
+
+Creating virtual build environments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Snapcraft instantiates and executes builds on self-allocated virtual instances. It uses
+the `Requests`_ library to install Multipass on Windows. Build environments for other
+operating systems handled with snapd, as described below.
+
+Communication with snapd
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Snapcraft uses the Requests library to communicate over Unix sockets with the local
+`snap daemon (snapd)`_. These requests are used to fetch information about required
```

Comment:
The first instance of snapd was earlier than this, so it should have earned the alias.

Reply:
https://github.com/canonical/snapcraft/pull/5478/commits/b86011f1e4714e8ec77af1eac46208a99c354aa1